src/Elewant/AppBundle/Controller/ShepherdController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Elewant\AppBundle\Controller;
+
+use Elewant\AppBundle\Entity\Herd;
+use Elewant\AppBundle\Repository\HerdRepository;
+use Elewant\UserBundle\Entity\User;
+use Elewant\UserBundle\Security\UserProvider;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @Route("/shepherd")
+ */
+class ShepherdController extends Controller
+{
+
+    /**
+     * @Route("/{username}", name="shepherd_admire_herd")
+     */
+    public function shepherdAdmireHerdAction($username)
+    {
+        try {
+            $user = $this->getUserByName($username);
+        } catch (UsernameNotFoundException $e) {
+            throw $this->createNotFoundException('The user does not exist');
```

Comment:
Maybe we should move throwing this exception to the `getUserByName()` method itself?

B.t.w. `getHerd()` doesn't throw an exception when no herd is found. Is that intentional?